### PR TITLE
removing the schema level check for InDirect write 

### PR DESCRIPTION
**Pull request by suryasoma**
removing the schema level check for InDirect write and enabling the option enableModeCheckForSchemaFields only for Direct write

### CODE REVIEW

File: spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java [hunk 1]
```diff
@@ -365,7 +392,33 @@ public void testIndirectWriteToBigQueryWithDiffInSchemaNullableFieldAndDisableMo
...
             testDataset.toString(), TestConstants.DIFF_IN_SCHEMA_DEST_TABLE_NAME);
     int numOfRows = (int) bq.query(QueryJobConfiguration.of(query)).getTotalRows();
-    assertThat(numOfRows).isEqualTo(1);
+    assertThat(numOfRows).isGreaterThan(0);
+  }
+
+  @Test
+  public void testInDirectWriteToBigQueryWithDiffInDescription() throws Exception {
+    assumeThat(writeMethod, equalTo(WriteMethod.INDIRECT));
+    spark.conf().set("temporaryGcsBucket", temporaryGcsBucket);
```
Comment (davidrabinowitz) [inline spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java:401]: don't set globally, put in the write options

File: spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java [hunk 2]
```diff
@@ -365,7 +392,33 @@ public void testIndirectWriteToBigQueryWithDiffInSchemaNullableFieldAndDisableMo
...
+        .format("bigquery")
+        .mode(SaveMode.Append)
+        .option("writeMethod", writeMethod.toString())
+        .save(testDataset + "." + TestConstants.DIFF_IN_SCHEMA_DEST_TABLE_NAME);
+    String query =
+        String.format(
+            "select * from %s.%s",
+            testDataset.toString(), TestConstants.DIFF_IN_SCHEMA_DEST_TABLE_NAME);
+    int numOfRows = (int) bq.query(QueryJobConfiguration.of(query)).getTotalRows();
+    assertThat(numOfRows).isGreaterThan(0);
```
Comment (davidrabinowitz) [inline spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java:421]: can we get an accurate number?

File: spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java [hunk 3]
```diff
@@ -317,31 +317,58 @@ public void testDirectWriteToBigQueryWithDiffInSchemaAndDisableModeCheck() throw
...
                 .mode(SaveMode.Append)
                 .option("writeMethod", writeMethod.toString())
-                .option("enableModeCheckForSchemaFields", true)
                 .save(testDataset + "." + TestConstants.DIFF_IN_SCHEMA_DEST_TABLE_NAME));
   }
 
+  @Test
+  public void testInDirectWriteToBigQueryWithDiffInSchemaAndModeCheck() throws Exception {
+    assumeThat(writeMethod, equalTo(SparkBigQueryConfig.WriteMethod.INDIRECT));
+    spark.conf().set("temporaryGcsBucket", temporaryGcsBucket);
```
Comment (davidrabinowitz) [inline spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java:350]: don't set globally, put in the write options

File: spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java [hunk 4]
```diff
@@ -317,31 +317,58 @@ public void testDirectWriteToBigQueryWithDiffInSchemaAndDisableModeCheck() throw
...
+        .mode(SaveMode.Append)
+        .option("writeMethod", writeMethod.toString())
+        .option("enableModeCheckForSchemaFields", true)
+        .save(testDataset + "." + TestConstants.DIFF_IN_SCHEMA_DEST_TABLE_NAME);
+    String query =
+        String.format(
+            "select * from %s.%s",
+            testDataset.toString(), TestConstants.DIFF_IN_SCHEMA_DEST_TABLE_NAME);
+    int numOfRows = (int) bq.query(QueryJobConfiguration.of(query)).getTotalRows();
+    assertThat(numOfRows).isGreaterThan(0);
```
Comment (davidrabinowitz) [inline spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java:369]: can we get an accurate number?

File: spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java [hunk 5]
```diff
@@ -317,31 +317,58 @@ public void testDirectWriteToBigQueryWithDiffInSchemaAndDisableModeCheck() throw
             "select * from %s.%s",
             testDataset.toString(), TestConstants.DIFF_IN_SCHEMA_DEST_TABLE_NAME);
     int numOfRows = (int) bq.query(QueryJobConfiguration.of(query)).getTotalRows();
-    assertThat(numOfRows).isEqualTo(1);
+    assertThat(numOfRows).isGreaterThan(0);
```
Comment (davidrabinowitz) [inline spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java:320]: can we get an accurate number?

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java (hunk 4 above)
Comment (suryasoma) [reply spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java]: done

File: spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java (hunk 3 above)
Comment (suryasoma) [reply spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java]: done

File: spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java (hunk 2 above)
Comment (suryasoma) [reply spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java]: done

File: spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java (hunk 1 above)
Comment (suryasoma) [reply spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java]: done

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**